CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Tracking/AnalyticsManager.swift

```diff
@@ -30,15 +31,17 @@ public class AnalyticsManager {
         }
     }
 
-    private static func initializeMixpanel(with deviceID: String) {
-        mixpanelInstance = Mixpanel.initialize(token: mixPanelToken,
+    private static func initializeMixpanel(with deviceID: String, token: String?) {
+        guard let token else { return }
+        mixpanelInstance = Mixpanel.initialize(token: token,
                                                trackAutomaticEvents: false,
                                                serverURL: "https://api-eu.mixpanel.com")
         mixpanelInstance?.identify(distinctId: deviceID)
     }
 
-    private static func initializeAmplitude(with deviceID: String) {
-        Amplitude.instance().initializeApiKey(amplitudeKey)
+    private static func initializeAmplitude(with deviceID: String, apiKey: String?) {
+        guard let apiKey else { return }
```

Comment:
[suggestion] we need to add this check `guard let apiKey else { return }` also where the `Amplitude.instance().logEvent...`  is called in order to fix this:
```
ERROR: apiKey cannot be nil or empty, set apiKey with initializeApiKey: before calling logEvent
```

Reply:
@ValentinaIancu-Gini I think its better to keep this error, because we will see that something is not configured correctly. Amplitude already has this check in `logEvent` :
```
    if (self.apiKey == nil) {
        AMPLITUDE_ERROR(@"ERROR: apiKey cannot be nil or empty, set apiKey with initializeApiKey: before calling logEvent");
        return;
    }
```

Reply:
@mrkulik anyway I'm thinking that this will not be a problem if we are doing same implementation for the queue of events for amplitude. Let's discuss this.